### PR TITLE
doc transition

### LINKED ISSUE #393: [Doc] Transition
How to use them,
How to create them.

### PR DESCRIPTION
**Pull request by quentinderoubaix**
closes #393

### CODE REVIEW

File: docs/01-Headless/04-Transitions.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Transitions
+
+## Overview
+
+Transitions in AgnosUI are widgets allowing to display / hide content smoothly through an animation. At its heart, a transition is a [Directive](03-Directives.md) managing the visibility of inner content by attaching / detaching classes.
```
Comment (andreascorti) [inline docs/01-Headless/04-Transitions.md:5]: `widgets` ? what do you mean by widget?

File: docs/01-Headless/04-Transitions.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Let's see this in action by implementing a **Collapse** component:
+
+```sample
+{Collapse example:transition/collapse:129}
+```
+
+AgnosUI provides several transitions to use (collapse vertical or horizontal, fade in/out) but you may create your own using the `createSimpleClassTransition` utility and a bit of css. Here is another example with a flip animation:
+
+```sample
+{Filp in example:transition/flip:93}
```
Comment (andreascorti) [inline docs/01-Headless/04-Transitions.md:19]: typo :) `Filp`

File: docs/01-Headless/04-Transitions.md (hunk 1 above)
Comment (quentinderoubaix) [reply docs/01-Headless/04-Transitions.md:5]: I mean AgnosUI widget, so controlled by props and returning stores.

Comment (andreascorti) [reply docs/01-Headless/04-Transitions.md:5]: It is a more global reflection on the term `widget` which is not ubiquitous even if we talk about widgets all the time :D

**Review by andreascorti — APPROVED** [review]
LGTM with minor comments :)

**Comment by quentinderoubaix** [comment]
Thanks for the review @andreascorti !

**Pull request merged**